suse_macros.in

```diff
@@ -315,3 +315,28 @@ Provides translations for the \"%{-n:%{-n*}}%{!-n:%{name}}\" package.
   FFLAGS="${FFLAGS:-%{?build_fflags}}" ; export FFLAGS ; \
   FCFLAGS="${FCFLAGS:-%{?build_fflags}}" ; export FCFLAGS ; \
   LDFLAGS="${LDFLAGS:-%{?build_ldflags}}" ; export LDFLAGS
+
+# Do the necessary checks in order to enable livepatching support.
+# Enable livepatching support for SLE15-SP4 onwards. It requires
+# compiler support introduced there.
+
+# Currently only supported on x86_64.
+
+# Expand to the necessary livepatching compilation flags if the macro
+# livepatching is defined in the spec file.
+%cflags_livepatching %{lua: if rpm.expand("%livepatchable") == "1" then if rpm.expand("%_arch") == "x86_64" then print("-fpatchable-function-entry=16,14 -fdump-ipa-clones") end end}
+
+# Ipa-clones are files generated by gcc which logs changes made across
+# functions, and we need to know such changes to build livepatches
+# correctly. These files are intended to be used by the livepatch
+# developers and may be retrieved by using `osc getbinaries`.
+%pack_ipa_dumps \
+  %if %{with livepatching} || 0%{?livepatchable} \
```

Comment:
Please remote here the `%{with livepatching}` leftover.

Reply:
@marxin done.